File: 17solve/17-151.pas
// Автор В.Н. Шубинкин
##
var a := ReadLines('17-1.txt').Select(t -> t.ToInteger).ToArray;
var pair_count := 0;
var minEl := 20000;
for var i := 0 to a.High - 1 do
  if (a[i] mod 3 = 0) and (abs(a[i]) mod 10 = 6)
      or (a[i + 1] mod 3 = 0) and (abs(a[i + 1]) mod 10 = 6) then
  begin
    pair_count += 1;
    minEl := min(minEl, a[i], a[i + 1])    
  end;
println(pair_count, minEl)

File: 17solve/17-151.py
a = [int(x) for x in open('17-1.txt')]

def valid( x ):
  return abs(x) % 10 == 6 and abs(x) % 3 == 0

count = 0
mi = 20000
for i in range(len(a) - 1):
  if valid(a[i]) or valid(a[i+1]):
    count += 1
    mi = min(mi, a[i], a[i + 1])
print(count, mi)
